dags/common/utils.py

```diff
@@ -203,28 +202,32 @@ def check_dagrun_state(dagrun: DagRun, not_allowed_states=[], allowed_states=[])
     return all(states_values)
 
 
-def process_zip_file(file_bytes, file_name):
+def process_zip_file(file_bytes, file_name, only_specific_file):
     file_bytes.seek(0)
     with zipfile.ZipFile(file_bytes) as zip:
         for filename in zip.namelist():
-            file_prefix = ".".join(file_name.split(".")[:-1])
+            if only_specific_file and only_specific_file not in filename:
+                continue
             zip_file_content = zip.read(filename)
+            file_prefix = ".".join(file_name.split(".")[:-1])
             s3_filename = os.path.join(file_prefix, filename)
             yield (zip_file_content, s3_filename)
 
 
-def process_tar_file(file_bytes, file_name):
+def process_tar_file(file_bytes, file_name, only_specific_file):
```

Comment:
I would make it optional `only_specific_file` :)

Reply:
@ErnestaP ☝️

Reply:
but it's optional already in the main function, from where the process_tar and zip is called:
https://github.com/cern-sis/workflows/blob/33024a61c390aeb8ec2c6610d073e5190c7b45e6/dags/common/utils.py#L229

Process_tar/zip is never called separately, its always called from process_archive
Should I make it still optional in these functions, in our case?

Reply:
yep otherwise, if we have to use the functions directly, we will always have to pass the specific file :)

Reply:
and actually we could just:

```

def process_archive(file_bytes, file_name, **kwargs):
```

and pass them down to the other functions for simplicity